packages/amplify-category-api/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.js

```diff
@@ -504,18 +533,63 @@ function validatePathName(name, paths) {
     return 'Each path part must use characters a-z A-Z 0-9 - and must not be empty.\nOptionally, a path part can be surrounded by { } to denote a path parameter.';
   }
 
-  const split = name.split('/').filter(sub => sub !== ''); // because name starts with a /, this filters out the first empty element
+  return true;
+}
+
+// checkForPathOverlap checks to see if the provided path name eclipses or overlaps any other paths in the provided list of paths.
+//
+// checkForPathOverlap returns false if the provided path name does not overlap with any of the other provided paths.
+// checkForPathOverlap returns an object with the following structure if the provided path name does overlap with any of the provided paths:
+// {
+//   higherOrderPath: string,
+//   lowerOrderPath: string,
+// }
+//
+// checkForPathOverlap assumes that all provided paths have previously been run through validatePathName().
```

Comment:
Kudos on adding extensive comments like this one! It makes things much easier to understand 🙌

Reply:
❤️ 💛💚💙💜